pkg/discovery/service_test.go

```diff
@@ -28,3 +28,34 @@ func TestCache_Refresh(t *testing.T) {
 		}
 	})
 }
+
+func TestCache_Graveyard(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	planet, err := testplanet.New(t, 1, 8, 0)
```

Comment:
Test against both backends postgres and sqlite.

Reply:
Done